File: Leetcode/PY/Leetcode566-重塑矩阵.py
from typing import List
class Solution:
    def matrixReshape(self, nums: List[List[int]], r: int, c: int) -> List[List[int]]:
        row = len(nums)
        col = len(nums[0])
        if row * col != r * c:
            return nums
        ans = [[0] * c for _ in range(r)]
        for i in range(r):
            for j in range(c):
                l = (i * c + j) // col
                r = (i * c + j) % col
                ans[i][j] = nums[l][r]
        return ans

File: Leetcode/Rust/Leetcode566-重塑矩阵.rs
impl Solution {
    pub fn matrix_reshape(nums: Vec<Vec<i32>>, r: i32, c: i32) -> Vec<Vec<i32>> {
        let row = nums.len();
        let col = nums[0].len();
        let (r,c) = (r as usize,c as usize);
        if col * row != c * r{
            return nums
        }
        let mut ans = vec![vec![0;c];r];
        for i in 0..r{
            for j in 0..c{
                let idx = (i * c + j);
                let (x,y) = (idx / col,idx % col);
                ans[i][j] = nums[x][y];
            }
        }
        ans
    }
}